lib/modules.nix

```
@@ -342,28 +343,52 @@ rec {
       };
 
       result = withWarnings {
+        _type = "configuration";
```

```suggestion
        _type = if specialArgs ? class then "${specialArgs.class}-evaluation" else "module-evaluation";
```

Or something along those lines, to be able to distinguish between the different evaluation results, maybe don't even set a type if there's no class.

---

This makes matching the type of all configurations harder. For example option docs generation should be able to match all of them.
In the status quo this is easy, as is writing a type checking function that checks both the type and the class.
I've picked "configuration" instead of "evaluation" because it matches the flake attribute conventions.